activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -264,6 +264,11 @@ def adapter_name
         self.class::ADAPTER_NAME
       end
 
+      # Does the database for this adapter exist?
+      def self.database_exists?(config)
```

Comment:
Do we want to expose this method in the api doc as public API?

Reply:
I thought this could be quite useful outside of the database tasks so I opted to keep it public, but I don't feel strongly about this I'm happy to make `:no doc:` it.